UN-OCHA/rwint9-site
# [RW-159] Rich editor

Ticket: RW-159

This enables CKEditor for the job/training body and how to apply/register fields with a custom field widget to convert from markdown to HTML (what CKEditor works with) and convert back to markdown (legacy content format in RW) when saving. 

The default CKeditor styling is good enough I think:

<img width="953" alt="Screen Shot 2021-12-09 at 18 57 04" src="https://user-images.githubusercontent.com/696348/145374663-cde40f1e-36e4-4c61-804e-54b2a256d1a4.png">

We could inject the subtheme styles.css to be more like the end result, @left23 what do you think?

## Testing

- Create a job/training
- Add some content to the body for example with some styles (bold, heading, list etc.)
- Save and check that the output seems coherent with what was entered.
- Edit the node, save again, check nothing changes (there should be no new entry in the revision history at the bottom of the page as nothing changed)
- Edit the node, copy and paste some content from various sources and check that what is displayed in the editor seems correct (only the following tags are preserved: `<p> <a> <em> <strong> <ul> <ol> <li> <h2> <h3> <h4> <h5> <h6>`).

## Issue

- RW-273: pressing enter when the modal to add a link is open redirects to the link form instead of closing the modal and updating the link. Also pressing tab never focus the modal button.
